### DISCUSSION TITLE
Will OpenTofu be supported?

### DISCUSSION THREAD

**Discussion by CodeBleu** [coder/coder#9957] [Feature Request]
With all the Hashicorp Terraform drama, I was wondering if there are plans to make sure OpenTofu will be supported with this project?

**Comment by smolinari** [comment]
Check this out.  https://github.com/coder/coder/discussions/9241

There is a link to a blog post too in that discussion, but to be sure you don't miss it. :grin: 

https://coder.com/blog/hashicorp-license

Scott

**Reply by CodeBleu** [reply]
Thanks, but im disappointed with that blog post.  Would have been nice to have seen Coder be more supportive of a true open source approach.  Hopefully there will be a change of heart to support a truly open source approach like OpenTofu.

**Reply by smolinari** [reply]
I'm on the fence and more about Hashicorp's change of license than Coder's decision. 

On the one side, I can understand Terraform's no-compete clause on their OSS software. They put a lot of effort into the software and they founded it and certain companies taking that work and using it for their own profit with absolutely no return to Hashicorp is basically stealing. And thus, preventing the stealing is a worthy cause IMHO. Had there been a culture of "I use your stuff that brings me a ton of money, I'll share that money with you" there might not be these kinds of no-compete licenses, which btw, are spreading. It might be the norm one day. 

On the other side, open source is about fostering community and collaboration around the code. I'm certain a lot of people outside of Hashicorp worked on Terraform too. For sure a good number of the providers are from third parties, which make Terraform so valuable to begin with. So, the Terraform in its complete form is not all Hashicorp's alone. They don't really have a 100% right to enforce such a license. It's like them offering everyone an egg. The community turned it into a golden goose and now Hashicorp want to keep the goose for themselves. 

So, Coder taking a neutral stance (in a sense) and just sticking with Hashicorp's software is understandable. 

I don't believe in "true or truly" open source software, because of the simple fact that honor is necessary for it to work and there are just too many people, possibly even the majority, who will drop any iota of honor in a heart beat for the chance to make money. It's sad, but true. 

Scott

**Comment by matifali** [comment]
See #12009 for a related discussion.
